File: 2460-apply-operations-to-an-array/main.go
func applyOperations(nums []int) []int {
    
    n := len(nums)
    for i := 0; i < n - 1; i++ {
        if nums[i] == nums[i+1] {
            nums[i] *= 2
            nums[i + 1] = 0
        }
    }

    l := 0
    for r := 0; r < n; r++ {
        if nums[r] != 0 {
            nums[r], nums[l] = nums[l], nums[r]
            l++
        }
    }

    return nums
}

File: 2460-apply-operations-to-an-array/main.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        
        N = len(nums)

        for i in range(N - 1):
            if nums[i] == nums[i + 1]:
                nums[i] *= 2
                nums[i + 1] = 0

        l = 0
        for r in range(N):
            if nums[r]:
                nums[r], nums[l] = nums[l], nums[r]
                l += 1

        return nums
